### ISSUE TITLE
Issues with sphinx `Locale` transform (assumes rST)

### ISSUE THREAD

**Issue by jpmckinney** [executablebooks/MyST-Parser#302]
![Screen Shot 2021-02-05 at 5 11 55 PM](https://user-images.githubusercontent.com/26463/107094633-5d8abb00-67d5-11eb-9d89-ba77098a9224.png)

Documentation repository: https://github.com/open-contracting/standard/pull/1197

To reproduce:

```
pip install -r requirements.txt
make
```

**Comment by jpmckinney** [comment]
cc @choldgraf

**Comment by chrisjsewell** [comment]
can you provide a minimal working of example of a file that causes this. (it may be related to https://github.com/executablebooks/MyST-Parser/issues/262)

**Comment by chrisjsewell** [comment]
also does this only happen for a translated document?

**Comment by jpmckinney** [comment]
Yes, it only happens for a translated document. I'll try to work up an example. As you can see in the screenshot, the warning text is where the Spanish-language header would be otherwise.

**Comment by chrisjsewell** [comment]
it looks to me that there is something weird going on with these translated texts, where it the heading levels are messed up:

For example I don't see why it would think that: https://raw.githubusercontent.com/choldgraf/standard/myst-parser-switch/docs/404.md, has a level 2 (i.e. `##`) level heading.

```
 /home/runner/work/standard/standard/docs/404.md:5:<translated>:1: WARNING: Non-consecutive header level increase; 0 to 2
```

Is there a way to view the source text of these special `<translated>` files?

**Comment by jpmckinney** [comment]
Here's a simple example: https://github.com/jpmckinney/myst-parser-tests/blob/master/index.md

Thanks for the quick responses!

**Comment by jpmckinney** [comment]
Note that, using the code in #303, the heading does appear in Spanish.

**Comment by chrisjsewell** [comment]
> Note that, using the code in #303, the heading does appear in Spanish.

indeed, but that is not addressing the root issue of the heading levels

I'll have to leave @choldgraf to look into this for now

(perhaps the translation is converting `#` to `##` for some reason 🤷)

**Comment by chrisjsewell** [comment]
So this is actually an upstream bug with the `Locale` transform: https://github.com/sphinx-doc/sphinx/blob/163c7bbdc12411818de1ce0204ff29cd911bcaf2/sphinx/transforms/i18n.py#L99,
which is a number of places assumes that the parser is the ReStructuredText one.

In particular, for `title` nodes, it adds an `-` underline (https://github.com/sphinx-doc/sphinx/blob/163c7bbdc12411818de1ce0204ff29cd911bcaf2/sphinx/transforms/i18n.py#L266),
which in Markdown just so happens to be a level two Setext header.

This leads to parsing e.g.:

```markdown
Un título
------------------
```

to

```xml
<document source="/Users/chrisjsewell/Documents/GitHub/myst-parser-intl-tests/index.md:6:<translated>">
    <system_message level="2" line="1" source="/Users/chrisjsewell/Documents/GitHub/myst-parser-intl-tests/index.md:6:<translated>" type="WARNING">
        <paragraph>
            Non-consecutive header level increase; 0 to 2
    <section ids="un-titulo" names="un\ título">
        <title>
            Un título
```

then the first node is assumed to be the title => "Non-consecutive header level increase; 0 to 2" becoming the title.

translations in literal block also look to be an issue

So yeh this requires a "fix" in the sphinx `Locale` code and/or overriding the transform in myst-parser (the first being ideal)

**Comment by jpmckinney** [comment]
Hmm, for my organization's project, we might need to do both, since even if there's a fix for Sphinx, it will only be available in the latest release (whenever that occurs).

@choldgraf Will you be having a look? This would be a blocker for using MyST. If we need to temporarily fork Sphinx or MyST-Parser, that's fine.

**Comment by jpmckinney** [comment]
I created https://github.com/sphinx-doc/sphinx/issues/8852, but I assume the issue will only be fixed in Sphinx>=4, which myst-parser doesn't presently support.

**Comment by jpmckinney** [comment]
Sphinx merged by PR in its 3.x branch to fix the heading level jump: https://github.com/sphinx-doc/sphinx/pull/8853

The maintainer suggests an approach in https://github.com/sphinx-doc/sphinx/issues/8852.

I think we can close the issue here, since it's a problem with upstream.

**Comment by choldgraf** [comment]
ah sounds great - thanks for the update @jpmckinney

**Comment by chrisjsewell** [comment]
> I think we can close the issue here, since it's a problem with upstream.

I wouldn't close
This though the nail the upstream issue has actually been closed:
https://github.com/sphinx-doc/sphinx/issues/8852#issuecomment-778561279
